### PR TITLE
Add Algorithms.common: Get the Day of the Week

### PR DESCRIPTION
**Pull request by ChiaSuen**
### Description
The algorithm can be used to Get the Day of the Week and check the Day of the Week from the date

### Checklist

- [x] An issue was first created **before** opening this pull request
- [x] The new code follows the [contribution guidelines](CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests to ensure that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: Algorithms/Common/DayofWeek.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        /// </summary>
+        /// <returns>Return string of day of week of the date</returns>
+        /// <param name="date">Date.</param>
+        public static string GetDayofWeek(DateTime date)
+        {
+
+            int count = 1; //Int to shows Ordinal number of occurrence of day of week
+            DateTime Occurrence = date.AddDays(-7); //Date to check the ordinal number of day of week 
+
+            for (string month = date.ToString("MMMM"); month == Occurrence.ToString("MMMM"); Occurrence = Occurrence.AddDays(-7))
```
Comment (dmitryvhf) [inline Algorithms/Common/DayofWeek.cs:19]: 1. Alot conversion without regional settings.
2. Alot string operations
3. Easy syntax - .ToString("MMMM") -> .Month

Example 1th refactoring
`for (int month = date.Month; month == Occurrence.Month; Occurrence = Occurrence.AddDays(-7))`
Example 2nd refactiong:
`int count = date.Day % 7;`

File: Algorithms/Common/DayofWeek.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+namespace Algorithms.Common
+{
+    public static class DayofWeek
+
+    {
+        /// <summary>
+        /// Determine day of week of the date
+        /// </summary>
+        /// <returns>Return string of day of week of the date</returns>
+        /// <param name="date">Date.</param>
```
Comment (dmitryvhf) [inline Algorithms/Common/DayofWeek.cs:12]: _param_ must be before _returns_

File: Algorithms/Common/DayofWeek.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            switch (count)
+            {
+                case 1:
+                    return count + "st " + date.ToString("ddd");
+                case 2:
+                    return count + "nd " + date.ToString("ddd");
+                case 3:
+                    return count + "rd " + date.ToString("ddd");
+                default:
+                    return count + "th " + date.ToString("ddd");
```
Comment (dmitryvhf) [inline Algorithms/Common/DayofWeek.cs:34]: 1. "nd", "st", etc - without regional support.

> <returns>Return string of day of week of the date</returns>
2. Critical! You must return only day of week. But why you return week number too?
Or remove this switch, or change method description.
Want check weeks - do new method for SRP (SOLID).

You method it not count day of week. This only convert day number to string.

File: Algorithms/Common/DayofWeek.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        /// </summary>
+        /// <returns>Return string of day of week of the date</returns>
+        /// <param name="date">Date.</param>
+        /// <param name="month">Month.</param>
+        /// <param name="year">Year.</param>
+        //overload
+        public static string GetDayofWeek(int date, int month, int year)
+        {
+            DateTime date2 = new DateTime(year, month, date);
+            return GetDayofWeek(date2);
```
Comment (dmitryvhf) [inline Algorithms/Common/DayofWeek.cs:51]: 1. not _date_, but _day_
2. Easy return:
`return GetDayofWeek(new DateTime(year, month, day));`

**Review by dmitryvhf — CHANGES_REQUESTED** [review]
Decline.
1. My opinition - this is not algorythm.
2. Alot code style problems

**Comment by ChiaSuen** [comment]
Thanks for your comment, will try to do it better next time

**Pull request closed without merging**